src/injectscripts/jobs/situational_awareness.js

```diff
@@ -1,7 +1,541 @@
+                let lat = fire.geometry.coordinates[1];
+                let lon = fire.geometry.coordinates[0];
+
+                let name = fire.properties.title;
+                let details = fire.properties.description;
+                let category = fire.properties.category;
+
+                let relativeIcon = rfsIcons[category] || rfsIcons['unknown'];
+                let icon = lighthouseUrl + relativeIcon;
+
+                console.debug(`RFS incident at [${lat},${lon}]: ${name}`);
+                mapLayer.addImageMarker(lat, lon, icon, name, details);
+                count++;
+            }
+        }
+    }
+    console.info(`added ${count} RFS incidents`);
+}
+
+/**
+ * Builds the params string for the helicopter's update request.
+ *
+ * @returns the params.
+ */
+function buildHeliParams() {
+    // Build the query url
+    let params = '';
+    for (let i = 0; i < aircraft.length; i++) {
+        params += i === 0 ? '?' : '&';
+        params += 'icao24=' + aircraft[i].icao24.toLowerCase();
+    }
+
+    return params
+}
+
+/**
+ * Shows the current position of known rescue helicopters on the map.
+ *
+ * @param mapLayer the map layer to add to.
+ * @param data the data to add to the layer.
+ */
+function showRescueHelicopters(mapLayer, data) {
+    console.info('showing rescue helicopters');
+
+    let count = 0;
+    if (data && data.states) {
+        for (let i = 0; i < data.states.length; i++) {
+            let icao24 = data.states[i][0];
+            let positionUpdated = data.states[i][3];
+            let lon = data.states[i][5];
+            let lat = data.states[i][6];
+            let alt = data.states[i][7];
+            let heading = data.states[i][10] || 0;
+
+            let updated = "unknown";
+            if (positionUpdated) {
+                updated = moment(positionUpdated * 1000).format('YYYY-MM-DD HH:mm:ss');
+            }
+            let dateDetails =
+                `<div class="dateDetails">\
+                 <div><span class="dateDetailsLabel">Last Position Update: </span> ${updated}</div>\
+                 </div>`;
+
+            let heli = findAircraftById(icao24);
+            let name = heli.name + ' ' + heli.rego;
+            let details =
+                `<div>${heli.model}</div>\
+                 <div>Lat: ${lat} Lon: ${lon} Alt: ${alt}</div>\
+                 ${dateDetails}`;
+
+            console.debug(`helo at [${lat},${lon}]: ${name}`);
+            let marker = mapLayer.createImageMarker(heli.getIcon(), name, details);
+            marker.setWidth(32);
+            marker.setHeight(32);
+            marker.setAngle(heading);
+            mapLayer.addMarker(lat, lon, marker);
+            count++;
+        }
+    }
+    console.info(`added ${count} rescue helicopters`);
+}
+
+/**
+ * Finds an aircraft by it's ICAO-24 ID.
+ *
+ * @param icao24 the ID.
+ * @returns the aircraft, or {@code null} if no match is found.
+ */
+function findAircraftById(icao24) {
```

Comment:
it really doesn't matter at this scale but worth considering storing the aircraft in a hash/dict indexed by their `icao24.toLowerCase()` so you don't have to iterate through to find one. If you don't want to copy/paste the icao24 twice that's fine just make it populate a new hash (`aircraftById`) at init so the iteration is done once and not every time you need to find an aircraft.